jiant/tasks/senteval_probing.py

```diff
@@ -0,0 +1,507 @@
+        self.path = path
+        self.max_seq_len = max_seq_len
+        self._label_namespace = self.name + "_tags"
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+
+    def get_all_labels(self):
+        return self.labels
+
+    def process_split(self, split, indexers, model_preprocessing_interface):
+        return process_single_pair_task_split(
+            split,
+            indexers,
+            model_preprocessing_interface,
+            label_namespace=self._label_namespace,
+            is_pair=False,
+            skip_indexing=False,
+        )
+
+    def get_sentences(self):
+        return self.sentences
+
+    def load_data(self):
+        """ Load data """
+
+        def load_json(data_file):
+            rows = pd.read_csv(data_file, encoding="ISO-8859-1")
+            labels = rows["1"].apply(lambda x: str(x.split("\t")[0]))
+            self.labels = list(set(labels.tolist()))
+            s1 = rows["1"].apply(lambda x: x.split("\t")[1])
+            s1 = s1.apply(
+                lambda x: tokenize_and_truncate(self._tokenizer_name, x, self.max_seq_len)
+            )
+            return s1.tolist(), [], labels.tolist(), list(range(len(rows)))
+
+        self.train_data_text = load_json(os.path.join(self.path, "train.tsv"))
+        self.val_data_text = load_json(os.path.join(self.path, "val.tsv"))
+        self.test_data_text = load_json(os.path.join(self.path, "test.tsv"))
+
+        sentences = []
+        for split in ["train", "val", "test"]:
+            split_data = getattr(self, "%s_data_text" % split)
+            sentences.extend(split_data[0])
+        self.sentences = sentences
+
+
+@register_task("se-probing-subj-number", rel_path="subj_number")
+class SEProbingSubjNumberTask(SingleClassificationTask):
+    """ Subjective number task """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super(SEProbingSubjNumberTask, self).__init__(name, n_classes=2, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+        self._label_namespace = self.name + "_tags"
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+
+    def get_all_labels(self):
+        return ["NN", "NNS"]
+
+    def process_split(self, split, indexers, model_preprocessing_interface):
+        return process_single_pair_task_split(
+            split,
+            indexers,
+            model_preprocessing_interface,
+            label_namespace=self._label_namespace,
+            is_pair=False,
+            skip_indexing=False,
+        )
+
+    def get_sentences(self):
+        return self.sentences
+
+    def load_data(self):
+        """ Load data """
+
+        def load_json(data_file):
+            rows = pd.read_csv(data_file, encoding="ISO-8859-1")
+            labels = rows["1"].apply(lambda x: str(x.split("\t")[0]))
```

Comment:
something seems off... Aren't these already .tsv's ? Why do we need to split on tabs again ?

Reply:
There are some datasets where the input is of the form  10\tWho knew who would be there ?, but is otherwise a valid csv (quite strange, but I just got around it with this method).